37728 internal aws

seongpil0948 opened the pull request:
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->
#### Description

<!-- Issue number (e.g. #1234) or full URL to issue, if applicable. -->
#### Link to tracking issue
Fixes #37728

<!--Describe what testing was performed and which tests were added.-->
#### Testing

<!--Describe the documentation added.-->
#### Documentation

<!--Please delete paragraphs that you did not use before submitting.-->

seongpil0948 committed:
Migrate internal/aws/awsutils to use aws-sdk-for-go-v2

mjlshen commented on internal/aws/awsutil/conn.go:
```diff
@@ -226,119 +206,40 @@ func ProxyServerTransport(logger *zap.Logger, config *AWSSessionSettings) (*http
-		}
+// getPartition returns AWS Partition for the provided region.
+func getPartition(region string) string {
+	// This is a simplified version - in production we would use the actual AWS SDK v2
```
We should definitely use the actual AWS SDK partitions package

seongpil0948 replied:
I will check if there are any other parts that I missed, not just that part. Thank you for the review.

seongpil0948 replied:
@mjlshen  AWS SDK v2 provides the GetPartition function in the internal module so, i worked in reference(https://github.com/aws/aws-sdk-go-v2/blob/main/internal/endpoints/awsrulesfn/partitions.go)

mjlshen commented on internal/aws/awsutil/conn.go:
```diff
@@ -109,90 +111,68 @@ func getProxyURL(finalProxyAddress string) (*url.URL, error) {
 }
 
+// GetAWSConfigSession maintains backwards compatibility with the original API but uses AWS SDK v2 internally
+func GetAWSConfigSession(ctx context.Context, logger *zap.Logger, cn ConnAttr, cfg *AWSSessionSettings) (aws.Config, error) {
```
Will anything use this function?

seongpil0948 replied:
that it is not. i will delete it

seongpil0948 replied:
removed it

seongpil0948 committed:
fix reflect mjlshen review

seongpil0948 committed:
enhancement(awsutil): conn partition

seongpil0948 committed:
feat: reflect the awsutil module to the component in use

seongpil0948 committed:
feat(awscontainerinsightreceiver): reflect the awsutil module to the component in use

seongpil0948 committed:
fix(awsxray): reflect the awsutil module to the component in use

seongpil0948 committed:
Merge branch 'main' into 37728_internal-aws

seongpil0948 committed:
fix(aws): resolve lint issues in AWS SDK v2 migration

This commit addresses several lint issues identified during the
AWS SDK v1 to v2 migration process:

1. Remove unused context parameters in AWS utility functions
2. Fix formatting issues with gci
3. Use appropriate assertion methods in tests

seongpil0948 committed:
Merge branch '37728_internal-aws' of github.com:seongpil0948/opentelemetry-collector-contrib into 37728_internal-aws

seongpil0948 committed:
Merge branch 'main' into 37728_internal-aws

seongpil0948 committed:
fix(awsxrayexporter): awsxray_test fail
BackendConnectionErrors.OtherCount was 1 in version 1, but HTTPCode4XXCount was 1 more after v2 conversion

seongpil0948 committed:
Merge branch 'main' into 37728_internal-aws
